src/main/scala/chiselverify/coverage/package.scala

```diff
@@ -386,9 +369,15 @@ package object coverage {
       * A value range that will be used for sampling
       * @param name the name of the value range that will be used to represent it in the report
       * @param range the actual scala range
+      * @param condition an extra condition that can be used to consider a hit
       */
-    case class Bins(name: String, range: Range) {
+    case class Bins(name: String, range: Range, condition: () => Boolean = () => true) {
         def ==(that: Bins): Boolean = (name == that.name) && (range.start == that.range.start) && (range.end == that.range.end)
+
+        def sample(portName: String, value: Int, coverageDB: CoverageDB): Unit =
+            if(range.contains(value) && condition()) coverageDB.addBinHit(portName, name, value)
+
+        def serialize: String = s"Bin( $name, $range ${ if(condition != (() => true)) condition })"
```

Comment:
@schoeberl The main changes are here, all I did was add the possibility to make a bin depend on an extra condition in order to consider a hit.

Everything else was simply reformatting @parzival3 's code to make it match the style used throughout the project.